lib/misc/blocks.coffee

```diff
@@ -0,0 +1,56 @@
+# TODO: docstrings
+
+scopes = require './scopes'
+
+module.exports =
+
+  getLine: (ed, l) ->
+    ed.getTextInBufferRange [[l, 0], [l, Infinity]]
+
+  isBlank: (l) -> l.match /^\s*(#.*)?$/
+  isEnd: (l) -> l.match /^end\b/
+  isCont: (l) -> l.match /^(else|elseif|catch|finally)\b/
+  isStart: (l) -> not (l.match(/^\s/) or @isBlank(l) or @isEnd(l) or @isCont(l))
```

Comment:
What about cases like
```julia
a(x) = 
x # cursor here, no indent
```
I mean, I know we can't reasonably do this without either relying on heuristics or calling `parse` on all lines, but we should probably note the cases we don't want to support somewhere...